src/root/components/ExportableView/index.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import html2canvas from 'html2canvas';
+import { startDownload } from '#utils/download-starter';
+
+import exportContext from './exportContext';
+
+function ExportableView(p) {
+  const [eventListeners, setEventListeners] = React.useState({});
+  const [containerRef, setContainerRef] = React.useState({ current: null });
+  const {
+    preview,
+    children,
+  } = p;
+
+  const startExport = React.useCallback(() => {
+    Object.values(eventListeners).forEach(listener => {
+      listener(true);
+    });
+
+    if (containerRef.current) {
+      // this is to skip current js event loop
+      html2canvas(containerRef.current, {useCORS: true}).then((renderedCanvas) => {
+        Object.values(eventListeners).forEach(listener => {
+          listener(false);
+        });
+
+        startDownload(
+          renderedCanvas,
+          `export-${new Date().getTime()}.png`,
```

Comment:
@frozenhelium it would be nice to be able to customize the filename for the exported file depending on context - is it possible to make the filename a prop (always appending datetime to the string sounds good).

Reply:
Sure, will do that